components/sysutils/net-snmp/net-snmp-demo.p5m

```diff
@@ -0,0 +1,201 @@
+    path=usr/demo/net-snmp/demo_module_7/Makefile
+file build/$(MACH32)/sun/sdk/demo/demo_module_7/README_demo_module_7 \
+    path=usr/demo/net-snmp/demo_module_7/README_demo_module_7
+file build/$(MACH32)/sun/sdk/demo/demo_module_7/SDK-DEMO6-MIB.txt \
+    path=usr/demo/net-snmp/demo_module_7/SDK-DEMO6-MIB.txt
+file build/$(MACH32)/sun/sdk/demo/demo_module_7/demo_module_7.c \
+    path=usr/demo/net-snmp/demo_module_7/demo_module_7.c
+file build/$(MACH32)/sun/sdk/demo/demo_module_7/demo_module_7.h \
+    path=usr/demo/net-snmp/demo_module_7/demo_module_7.h
+file build/$(MACH32)/sun/sdk/demo/demo_module_7/get_filesize \
+    path=usr/demo/net-snmp/demo_module_7/get_filesize
+file build/$(MACH32)/sun/sdk/demo/demo_module_7/register_file \
+    path=usr/demo/net-snmp/demo_module_7/register_file
+file build/$(MACH32)/sun/sdk/demo/demo_module_7/unregister_file \
+    path=usr/demo/net-snmp/demo_module_7/unregister_file
+file build/$(MACH32)/sun/sdk/demo/demo_module_8/Makefile \
+    path=usr/demo/net-snmp/demo_module_8/Makefile
+file build/$(MACH32)/sun/sdk/demo/demo_module_8/README_demo_module_8 \
+    path=usr/demo/net-snmp/demo_module_8/README_demo_module_8
+file build/$(MACH32)/sun/sdk/demo/demo_module_8/SDK-DEMO1-MIB.txt \
+    path=usr/demo/net-snmp/demo_module_8/SDK-DEMO1-MIB.txt
+file build/$(MACH32)/sun/sdk/demo/demo_module_8/demo_module_8.c \
+    path=usr/demo/net-snmp/demo_module_8/demo_module_8.c
+file build/$(MACH32)/sun/sdk/demo/demo_module_8/getme1LoadGroup \
+    path=usr/demo/net-snmp/demo_module_8/getme1LoadGroup
+file build/$(MACH32)/sun/sdk/demo/demo_module_8/me1LoadGroup.c \
+    path=usr/demo/net-snmp/demo_module_8/me1LoadGroup.c
+file build/$(MACH32)/sun/sdk/demo/demo_module_8/me1LoadGroup.h \
+    path=usr/demo/net-snmp/demo_module_8/me1LoadGroup.h
+file build/$(MACH32)/sun/sdk/demo/demo_module_9/Makefile \
+    path=usr/demo/net-snmp/demo_module_9/Makefile
+file build/$(MACH32)/sun/sdk/demo/demo_module_9/README_demo_module_9 \
+    path=usr/demo/net-snmp/demo_module_9/README_demo_module_9
+file build/$(MACH32)/sun/sdk/demo/demo_module_9/SDK-DEMO9-MIB.txt \
+    path=usr/demo/net-snmp/demo_module_9/SDK-DEMO9-MIB.txt
+file build/$(MACH32)/sun/sdk/demo/demo_module_9/demo_module_9.c \
+    path=usr/demo/net-snmp/demo_module_9/demo_module_9.c
+file build/$(MACH32)/sun/sdk/demo/demo_module_9/demo_module_9.h \
+    path=usr/demo/net-snmp/demo_module_9/demo_module_9.h
+file build/$(MACH32)/sun/sdk/demo/demo_module_9/get_demo_module_9 \
+    path=usr/demo/net-snmp/demo_module_9/get_demo_module_9
+file build/$(MACH32)/sun/sdk/demo/demo_module_9/set_demo_module_9 \
+    path=usr/demo/net-snmp/demo_module_9/set_demo_module_9
+file build/$(MACH32)/sun/sdk/demo/demo_module_9/walk_demo_module_9 \
+    path=usr/demo/net-snmp/demo_module_9/walk_demo_module_9
+license COPYING license="BSD, BSD-like"
+depend type=optional \
+    fmri=system/management/snmp/net-snmp/documentation@5.7.3
```

Comment:
The idea is the following. This should be an optional dependency on package version from history file. And history file causes renamed package to be published.
Now we have system/management/snmp/net-snmp/documentation package with obsolete and rename (to  system/management/snmp/net-snmp/demo) options. And system/management/snmp/net-snmp/demo package depending on the exact system/management/snmp/net-snmp/documentation version which contains this rename option.
So, when user installs net-snmp/demo package, it will trigger update of net-snmp/documentation package (if it's installed) to renamed version, which leads to its deinstallation.
I hope my explanation was sensible :) If not, look at https://docs.openindiana.org/dev/pdf/ips-dev-guide.pdf , chapter 6, dependency types.

Reply:
Great explanation, got it.